crates/net/downloaders/src/file_client.rs

```diff
@@ -452,28 +523,79 @@ impl ChunkedFileReader {
     /// Reads bytes from file and buffers as next chunk to decode. Returns byte length of next
     /// chunk to read.
     async fn read_next_chunk(&mut self) -> Result<Option<u64>, io::Error> {
-        if self.file_byte_len == 0 && self.chunk.is_empty() {
-            // eof
+        if !self.is_gzip && self.file_byte_len == 0 && self.chunk.is_empty() {
+            // eof for non-gzipped files
             return Ok(None)
         }
 
         let chunk_target_len = self.chunk_len();
         let old_bytes_len = self.chunk.len() as u64;
 
+        // If we already have enough data in the chunk, return it
+        if old_bytes_len >= chunk_target_len {
+            return Ok(Some(old_bytes_len))
+        }
+
+        // For gzipped files, if we have no data in chunk, try to read some to check EOF
+        if self.is_gzip && self.chunk.is_empty() {
```

Comment:
not a fan of all these gzip conditionals

both file variants now support the same interface so we should be able to make this work by only relying in the read trait and not use different read variants